### PR TITLE
Missing api

### LINKED ISSUE #24: Implement missing API features
As discussed in https://github.com/LukasBombach/de1/issues/20

```ts
de1.getMixTemp();
de1.getHeadTemp();
de1.getSteamTemp();
de1.getShotSettings();
de1.setShotSettings();
de1.getVersions();
de1.get(name: ConverterName);
de1.set(name: ConverterName, value: ConverterValue);
```

### PR DESCRIPTION
**Pull request by LukasBombach**
Implements https://github.com/LukasBombach/de1/issues/24 (fix https://github.com/LukasBombach/de1/issues/24) and adds

```ts
de1.getMixTemp();
de1.getHeadTemp();
de1.getSteamTemp();
de1.getShotSettings();
de1.setShotSettings();
de1.getVersions();
de1.get(name: ConverterName);
de1.set(name: ConverterName, value: ConverterValue);
```

- [x] Implementation
- [x] Types
- [ ] Tests

### CODE REVIEW

**Comment by LukasBombach** [comment]
I had to write fix https://github.com/LukasBombach/de1/issues/24 because then this PR will close the ticket automatically 🙄

File: packages/de1/src/de1.ts [hunk 1]
```diff
@@ -75,11 +75,58 @@ export default class DE1 {
...
     this.machine.off(name, listener);
   }
+
+  private async mergeCurrentValue<N extends Name>(
+    name: N,
+    value: Partial<Value<N>>,
+  ): Promise<Value<N>> {
+    if (typeof value !== "object" || value === null) return value;
+    const currentValue = await this.machine.read(name);
+    return Object.assign({}, currentValue, value);
```
Comment (oscar-b) [inline packages/de1/src/de1.ts:145]: I prefer this syntax, but up to you:

```suggestion
    return {...currentValue, ...value};
```

Comment (LukasBombach) [reply packages/de1/src/de1.ts:145]: No you're absolutely right!

**Comment by LukasBombach** [comment]
I have decided to go back to Object.assign. This is one of the "I have TypeScript" moments, but in the end, TypeScript is right. What I am doing there is not typesafe. But the good news is, that there is an implementation to validate user input for the things we send to the de1

**Comment by LukasBombach** [comment]
Hey @oscar-b I am flagging this as ready for review. There are some tests missing, but because of my current situation, I'd say "let's go" and see what we have to fix later.

**Review by oscar-b — APPROVED** [review]
Looks good to me, but I’m on mobile and haven’t been able to test anything myself. 

Did you consider using something like `yup` for runtime validation?

File: packages/de1/src/converters/shotDescriptionHeader.ts [hunk 2]
```diff
@@ -20,19 +20,30 @@ function decode(data: Buffer): ShotDescriptionHeader {
...
+    .char(header.numberOfFrames)
+    .char(header.numberOfPreinfuseFrames)
+    .char(header.minimumPressure * 16)
+    .char(header.maximumFlow * 16).buffer;
 }
 
-const converter: Converter<ShotDescriptionHeader> = {
+function validate(header: Partial<ShotDescriptionHeader>): boolean {
+  return Serializer.validate(header, {
+    headerV: Number(),
```
Comment (oscar-b) [inline packages/de1/src/converters/shotDescriptionHeader.ts:34]: Why `Number()` and not just `’number’`, matching `typeof` results?

Comment (LukasBombach) [reply packages/de1/src/converters/shotDescriptionHeader.ts:34]: to be honest, that's probably just bad code

**Comment by LukasBombach** [comment]
Hey, I'm just gonna merge this now like what the hell. Let's just fix stuff.

**Comment by LukasBombach** [comment]
So we can go on and start with the app

**Pull request merged**